quantum/wear_leveling/wear_leveling.c

```diff
@@ -152,8 +152,35 @@
 static struct __attribute__((__aligned__(BACKING_STORE_WRITE_SIZE))) {
     __attribute__((__aligned__(BACKING_STORE_WRITE_SIZE))) uint8_t cache[WEAR_LEVELING_LOGICAL_SIZE];
     uint32_t                                                       write_address;
+    bool                                                           unlocked;
 } wear_leveling;
 
+/**
+ * Locking helper: unlock
+ */
+static inline bool wear_leveling_unlock(void) {
+    if (wear_leveling.unlocked) {
+        return false;
+    }
+    if (backing_store_unlock()) {
+        wear_leveling.unlocked = true;
+    }
+    return wear_leveling.unlocked;
+}
+
+/**
+ * Locking helper: lock
+ */
+static inline bool wear_leveling_lock(void) {
+    if (!wear_leveling.unlocked) {
+        return false;
+    }
+    if (backing_store_lock()) {
+        wear_leveling.unlocked = false;
+    }
+    return !wear_leveling.unlocked;
+}
```

Comment:
I wonder if this should be tri-state functions?

1. Already locked/unlocked - this can either be an error in the internal state of the wear leveling or actually correct
2. Not locked/unlocked but attempting to failed by hardware - always a fatal error
3. lock/unlock succeeds - happy path